### DISCUSSION TITLE
Strip flow types from production builds

### DISCUSSION THREAD

**Discussion by bacarybruno** [react-native-community/discussions-and-proposals#810] [Ideas]
Heyo all 👋

As you might know, React Native ships with Flow code included when it's installed. I'm curious whether this was intentional for a specific reason. Does it provide some advantages or benefits that we should be aware of?

I'm asking because stripping flow types can simplify the usage of other tools like `@swc/jest`.

Thanks 🙏

**Comment by bacarybruno** [comment]
Hello again 👋 I'm willing to work on it if needed. I just need to know if it was intentional to ship the flow code or not 🙏

**Comment by jvenene** [comment]
I'm not sure if this is about Flow types in production builds or being shipped with React Native.

> Strip flow types from production builds

I think Babel strips Flow types at build time, so there wouldn't be any Flow types in production builds.

> React Native ships with Flow code included when it's installed.

Are you referring to the Flow types being included when installing React Native from NPM? If so, it seems that is no longer the case in 0.81: https://github.com/facebook/react-native/commit/38acb4c0746e48ebb10729360788e26454736d1b.

**Reply by bacarybruno** [reply]
> Are you referring to the Flow types being included when installing React Native from NPM?

Yes, that's what I'm referring to 👍

> It seems that is no longer the case in 0.81

It seems that we still see quite a few files containing Flow code. For example, the index.js entrypoint imports index.js.flow. Is it planned to strip the flow code from these files before publishing the react-native package on NPM?

**Comment by bacarybruno** [comment]
Here is the formal proposal https://github.com/react-native-community/discussions-and-proposals/pull/949. Let's see where it leads us 🤞
